digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadwayDAO.scala

```diff
@@ -1080,32 +1080,78 @@ class RoadwayDAO extends BaseDAO {
         * */
 
-      s"""$query FROM (SELECT ely, road_number, road_part_number, start_addr_m, start_date, track, ROW_NUMBER() OVER (ORDER BY road_number, road_part_number, start_addr_m, track) AS numb
-                        FROM roadway r
-                        WHERE NOT EXISTS
-                          (SELECT 1 FROM roadway r2 WHERE r2.end_addr_m = r.start_addr_m AND r2.road_number = r.road_number AND r2.road_part_number=r.road_part_number AND r2.track = r.track AND r2.start_date = r.start_date AND r2.valid_to IS NULL AND r2.end_date IS NULL)
-                        AND r.valid_to IS NULL AND r.end_date IS NULL $dateCondition $elyCondition $roadNumberCondition $roadPartCondition) mins
-                   JOIN (SELECT ely, road_number, road_part_number, end_addr_m, start_date, track, ROW_NUMBER() OVER (ORDER BY road_number, road_part_number, end_addr_m, track) AS numb
-                              FROM roadway r
-                              WHERE NOT EXISTS
-                                (SELECT 1 FROM roadway r2 WHERE r2.start_addr_m = r.end_addr_m AND r2.road_number = r.road_number AND r2.road_part_number=r.road_part_number AND r2.track=r.track AND r2.start_date = r.start_date AND r2.valid_to IS NULL AND r2.end_date IS NULL)
-                              AND r.valid_to IS NULL and r.end_date IS NULL $dateCondition $elyCondition $roadNumberCondition $roadPartCondition) maxs ON mins.numb = maxs.numb
-                  ORDER BY mins.road_number, mins.road_part_number, mins.start_addr_m""".stripMargin
+      s"""WITH roadways
+         |     AS (SELECT *
+         |         FROM   roadway r
+         |         WHERE  r.valid_to IS NULL AND r.end_date IS NULL $dateCondition $elyCondition $roadNumberCondition $roadPartCondition $query""".stripMargin
     }
 
     def fetchRoads(queryFilter: String => String): Seq[RoadForRoadAddressBrowser] = {
       val query =
-        """
-      SELECT mins.ely, mins.road_number, mins.track, mins.road_part_number, mins.start_addr_m,
-       maxs.end_addr_m, maxs.end_addr_m - mins.start_addr_m AS length, mins.start_date
-      """
+        """),
+          |     roadwayswithstartaddr
+          |     AS (SELECT ely,
+          |                road_number,
+          |                road_part_number,
+          |                start_addr_m,
+          |                start_date,
+          |                track,
+          |                Row_number()
+          |                  OVER (
+          |                    ORDER BY road_number, road_part_number, track, start_addr_m)
+          |                AS
+          |                numb
+          |         FROM   roadways r
+          |         WHERE  NOT EXISTS (SELECT 1
+          |                            FROM   roadways r2
+          |                            WHERE  r2.start_date = r.start_date
```

Comment:
...miks start_date tarttee olla NOT EXISTSin ehdoissa?

Reply:
se liittyy juurikin siihen pihviin, eli halutaan muodostaa roadway riveistä homogeenisiä pätkiä tie nuomeron, tieosan, elyn, ajoradan ja _alkupäivämäärän_ mukaan